Source/WarpX.cpp

```diff
@@ -1461,6 +1453,29 @@ WarpX::AllocLevelData (int lev, const BoxArray& ba, const DistributionMapping& d
         WarpX::fft_do_time_averaging,
         this->refRatio());
 
+
+#ifdef AMREX_USE_EB
+#ifdef WARPX_DIM_3D
+        amrex::Vector<int> fs_guard{guard_cells.ng_FieldSolver[0],
+                                guard_cells.ng_FieldSolver[1],
+                                guard_cells.ng_FieldSolver[2]};
+#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+        amrex::Vector<int> fs_guard{guard_cells.ng_FieldSolver[0],
+                                    guard_cells.ng_FieldSolver[1],
+                                    guard_cells.ng_FieldSolver[1]};
+#elif defined(WARPX_DIM_1D_Z)
+        amrex::Vector<int> fs_guard{guard_cells.ng_FieldSolver[0],
+                                guard_cells.ng_FieldSolver[0],
+                                guard_cells.ng_FieldSolver[0]};
+#endif
+        m_field_factory[lev] = amrex::makeEBFabFactory(Geom(lev), ba, dm,
+                                                       fs_guard,
+                                                       amrex::EBSupport::full);
+#else
```

Comment:
@WeiqunZhang @RemiLehe  `makeEBFabFactory` always takes a three-dimensional vector for the number of guard cells, no matter the dimensions we use to compile AMReX, while the number of components of guard_cells.ng_FieldSolver depends on the dimensions.
I couldn't find a better solution than the workaround I put in place above, but I am not sure it is ideal.